server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/RabbitMQJmapExtension.java

```diff
@@ -0,0 +1,127 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.jmap.rabbitmq;
+
+import java.io.IOException;
+
+import org.apache.james.CassandraRabbitMQJamesServerMain;
+import org.apache.james.DockerCassandraRule;
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.backend.rabbitmq.DockerRabbitMQSingleton;
+import org.apache.james.backends.es.EmbeddedElasticSearch;
+import org.apache.james.jmap.methods.integration.JamesWithSpamAssassin;
+import org.apache.james.jmap.methods.integration.SpamAssassinModule;
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.james.mailbox.store.search.PDFTextExtractor;
+import org.apache.james.modules.TestESMetricReporterModule;
+import org.apache.james.modules.TestElasticSearchModule;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.apache.james.modules.TestRabbitMQModule;
+import org.apache.james.modules.TestSwiftBlobStoreModule;
+import org.apache.james.server.core.configuration.Configuration;
+import org.apache.james.spamassassin.SpamAssassinExtension;
+import org.apache.james.util.Runnables;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.rules.TemporaryFolder;
+
+public class RabbitMQJmapExtension implements BeforeAllCallback, AfterAllCallback,
+    BeforeEachCallback, AfterEachCallback, ParameterResolver {
+
+    private static final int LIMIT_TO_20_MESSAGES = 20;
+
+    private final TemporaryFolder temporaryFolder;
+    private final DockerCassandraRule cassandra;
+    private final EmbeddedElasticSearch elasticSearch;
+    private final SpamAssassinExtension spamAssassinExtension;
+    private JamesWithSpamAssassin james;
+
+    public RabbitMQJmapExtension() {
+        this.temporaryFolder = new TemporaryFolder();
+        this.cassandra = new DockerCassandraRule();
+        this.elasticSearch = new EmbeddedElasticSearch(temporaryFolder);
+        this.spamAssassinExtension = new SpamAssassinExtension();
+    }
+
+    private JamesWithSpamAssassin james() throws IOException {
+        Configuration configuration = Configuration.builder()
+            .workingDirectory(temporaryFolder.newFolder())
+            .configurationFromClasspath()
+            .build();
+        return new JamesWithSpamAssassin(
```

Comment:
I didn't catch `JamesWithSpamAssassin` introduction but it looks so useless when you can register extensions instead

Reply:
I will introduce it in another PR, because of this PR is getting bigger. Using register extension will involve to edit the contract and it impls. Produces many changes.

Reply:
ok to do that into another PR but please don't close the issue until it's fixed

Reply:
#1926 PR is here, I will take care RabbitMQ + Swift + Cassandra + SpamAssassin Test after this PR is merged